### PR TITLE
Add more front end code standards

### PR DESCRIPTION
**Pull request by seancolsen**

### CODE REVIEW

**Comment by seancolsen** [comment]
@pavish Can you review this?

File: engineering/architecture/front-end-standards.md [hunk 1]
```diff
@@ -77,4 +77,219 @@ dateCreated: 2022-01-14T19:02:19.924Z
...
+
+Notes:
+
+- Not all of our components adhere to this guideline yet.
+
+
+## `type` vs `interface`
+
+- Prefer `interface` when possible.
+- Use `type` when necessary.
```
Comment (seancolsen) [inline engineering/architecture/front-end-standards.md:236]: @pavish this is some content that we haven't discussed before, but from reading your code I've deduced that we're on the same page about this (which is also the direction that I think the large TS community is going).

File: engineering/architecture/front-end-standards.md [hunk 2]
```diff
@@ -77,4 +77,219 @@ dateCreated: 2022-01-14T19:02:19.924Z
...
+
```
Comment (seancolsen) [inline engineering/architecture/front-end-standards.md:237]: @pavish this is some content that we haven't discussed before, but from reading your code I've deduced that we're on the same page about this (which is also the direction that I think the large TS community is going).

**Review by pavish — APPROVED** [review]
@seancolsen Looks good to me!

**Pull request merged**